### PR TITLE
Ef update 0.7

### PR DESCRIPTION
**Pull request by evanfields**
Pretty mindless update to pass tests and remove deprecation warnings on Julia 0.7. This is my first attempt at upgrading a public package, so advance apologies for any errors.

### CODE REVIEW

File: src/Loess.jl [hunk 1]
```diff
@@ -223,10 +223,10 @@ Args:
   `xs`: a matrix of predictors
   `q`: cut the ends of at quantiles `q` and `1-q`
 
-Modifies:
+odifies:
```
Comment (tlnagy) [inline src/Loess.jl]: typo?

File: src/kd.jl [hunk 2]
```diff
@@ -42,18 +39,18 @@ Returns:
...
 
-    bounds = Array{T}(2, m)
+    bounds = Array{T}(undef, 2, m)
     for j in 1:m
-	col = xs[:,j]
-	bounds[1, j] = minimum(col)
-	bounds[2, j] = maximum(col)
+	    col = xs[:,j]
+	    bounds[1, j] = minimum(col)
+	    bounds[2, j] = maximum(col)
```
Comment (tlnagy) [inline src/kd.jl]: Why the increase in indentation here?

File: REQUIRE [hunk 3]
```diff
@@ -1,4 +1,3 @@
-julia 0.5
-Compat 0.17
+julia 0.7-
 Distances
 IterTools
```
Comment (tlnagy) [inline REQUIRE]: We can drop the dependency on `IterTools` still Base now has `Iterators.product`

File: src/Loess.jl [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
 
-module Loess
 
-using Compat
+module Loess
 
-import IterTools.product
 import Distances.euclidean
 
+using Statistics
```
Comment (tlnagy) [inline src/Loess.jl]: What package is providing this module? Where is this used?

File: src/kd.jl [hunk 5]
```diff
@@ -169,16 +166,16 @@ function build_kdtree{T}(xs::AbstractMatrix{T},
...
-	if i == j
-	    coords[i] = [med]
-	else
-	    coords[i] = bounds[:, i]
-	end
+		if i == j
+			coords[i] = [med]
+		else
+			coords[i] = bounds[:, i]
+		end
```
Comment (tlnagy) [inline src/kd.jl]: What's wrong with the original indentation?

File: src/kd.jl (hunk 2 above)
Comment (evanfields) [reply src/kd.jl]: Indented loop bodies seems to be the Julia community standard (and I think increases readability) so in places where a diff was already involved, I figured may as well clean up a bit. Happy to keep the indents as-is if you prefer.

File: src/Loess.jl (hunk 4 above)
Comment (evanfields) [reply src/Loess.jl]: It's a standard library package providing functions such as `mean`. (Incidentally, this seems to be a change between 0.7-beta and 0.7-nightly, which is why the continuous integration tests are passing nightly but failing beta.)

Comment (tlnagy) [reply src/Loess.jl]: Ah. I'm still on 0.7.0-beta.0

File: src/kd.jl (hunk 2 above)
Comment (tlnagy) [reply src/kd.jl]: The standard indentation is 4 spaces, correct? It looks like original indentation was correct, no?

Comment (evanfields) [reply src/kd.jl]: I see what happened: it's a classic tabs-spaces mismatch. In the original code, the loop header is indented by 4 spaces and the body by 1 tab. Github renders the tab as 8 spaces so the body looks more indented than the header; my editor renders the tab as 4 spaces so it looked like the body needed more indentation. Turns out the entire package uses a mix of tabs and spaces so I will restore the original indentation here and perhaps at some point there will be a formatting PR.

**Comment by evanfields** [comment]
Hey @tlnagy just checking in to see if there are any other changes you want made. Thanks!

**Comment by tlnagy** [comment]
I think the only thing is that the lower bound in REQUIRE should technically be v0.7.0-beta2, but we could avoid this by not tagging this just yet.

**Comment by tlnagy** [comment]
Thanks for taking the time to do this!

**Comment by evanfields** [comment]
Just trying to learn and contribute a little to the community :) Next up, CSV.jl! Well, not really, I looked but think that's above my skill level. But if you have a suggestion of what to upgrade next I'm ears.

**Pull request merged**